### PR TITLE
Move utils to their own module

### PR DESCRIPTION
**Pull request by lboyette-okta**
bacon: test
Resolves: OKTA-89650

@rchild-okta

### CODE REVIEW

File: lib/cookies.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+  var cookieText = name + '=' + value + ';' + expiresText;
+  document.cookie = cookieText;
+
+  return cookieText;
+}
+
+function getCookie(name) {
+  var pattern = new RegExp(name + '=([^;]*)'),
+    matched = document.cookie.match(pattern);
```
Comment (rchild-okta) [inline lib/cookies.js]: nit: spacing

File: lib/cookies.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
+var util = require('./util');
+
+function setCookie(name, value, expiresAt) {
+  var expiresText = '';
+  if (expiresAt) {
+    expiresText = ' expires=' + util.isoToDate(expiresAt).toUTCString() + ';';
```
Comment (rchild-okta) [inline lib/cookies.js]: Do we ever care about having it in date format? How about just making the util function isoToUTCString?

File: lib/index.js [hunk 3]
```diff
@@ -190,44 +46,8 @@ function getLink(obj, linkName, altName) {
...
-  }
-}
-
-function deleteCookie(name) {
-  setCookie(name, '', '1970-01-01T00:00:00Z');
-}
-
 function addStateToken(res, options) {
-  var builtArgs = clone(options) || {};
+  var builtArgs = util.clone(options) || {};
```
Comment (rchild-okta) [inline lib/index.js]: It might be interesting in the future to just use lodash's per method packages for these types of methods:
https://www.npmjs.com/browse/keyword/lodash-modularized

File: lib/util.js [hunk 4]
```diff
@@ -48,9 +48,141 @@ function base64UrlDecode(str) {
...
+    }
+  }
+}
+
 module.exports = {
   base64UrlToBase64: base64UrlToBase64,
   base64UrlToString: base64UrlToString,
   stringToBuffer: stringToBuffer,
-  base64UrlDecode: base64UrlDecode
+  base64UrlDecode: base64UrlDecode,
```
Comment (rchild-okta) [inline lib/util.js]: We don't need to do it in this PR, but at some point we should add unit tests for each of these modules. I see our other tests as the "functional" tests, and we should have a separate set to exercise more of the edge cases in each of these individual functions.

^^ Another reason to move to just pulling a bunch these in from lodash (put the responsibility of fully vetting these edge cases in their test suites).

**Comment by rchild-okta** [comment]
A couple minor comments, but overall looks good - 🚀  after updating

File: lib/cookies.js (hunk 2 above)
Comment (lboyette-okta) [inline lib/cookies.js]: We don't use isoToDate anywhere else. I think I wrote this at a time when I wasn't sure what was necessary for OAuth. I can consolidate it.

File: lib/index.js (hunk 3 above)
Comment (lboyette-okta) [inline lib/index.js]: It's worth looking into

File: lib/util.js (hunk 4 above)
Comment (lboyette-okta) [inline lib/util.js]: Sounds good. It might justify pulling these base64 functions out into their own npm module
